index/scorch/merge.go

```diff
@@ -59,8 +58,13 @@ OUTER:
 				// lets get started
 				err := s.planMergeAtSnapshot(ourSnapshot, mergePlannerOptions)
 				if err != nil {
-					s.fireAsyncError(fmt.Errorf("merging err: %v", err))
 					_ = ourSnapshot.DecRef()
```

Comment:
hmm, not sure if this is correct either...

in the old behavior, the refcount was held, the error callback is fired, and then the ref-count is released.

in the proposed change, 1) the refcount is released, 2) then the error callback is fired.  If the callback depends on the index still being alive, the refcount release at step #1 might inadvertently cause trouble.

Reply:
fireAsyncError just logs a fatal event right now, but I see you point if that were to change.